### PR TITLE
Fix php doc, fix types mismatches

### PR DESCRIPTION
**Pull request by ZRazor**
I am targeting this branch, because it's a BC break.

## Changelog

```markdown
### Fixed
- Fixed phpdoc
- Added missing params to `ProductManagerInterface` methods
- Added missing `parent::__construct` call in `BaseBasket`
- Replaced string types with FQCN in `BaseProductProvider`
- Fixed `Transformer/Pool` usage of indefined `methods` property
```

- Didn't touch Api controllers, they need a separate fix.

### CODE REVIEW

**Comment by jordisala1991** [comment]
Can you explain which part is BC break? (it could be useful to isolate it maybe on a separate PR for master)

Another issue is with code comments, @greg0ire is working on comment removal for some bundles, we might need his help or look his PR to understand which comments we want to mantain and which not. The first one that comes to my mind is the inherit doc, he removed all of them from some bundles already.

Good work btw :)

File: src/BasketBundle/DependencyInjection/SonataBasketExtension.php [hunk 1]
```diff
@@ -65,8 +65,8 @@ public function load(array $configs, ContainerBuilder $container): void
     }
 
     /**
-     * @param \Symfony\Component\DependencyInjection\ContainerBuilder $container
-     * @param $config
+     * @param ContainerBuilder $container
+     * @param array            $config
      */
```
Comment (greg0ire) [inline src/BasketBundle/DependencyInjection/SonataBasketExtension.php]: Can be removed

File: src/BasketBundle/Entity/BaseBasketElement.php [hunk 2]
```diff
@@ -19,24 +19,20 @@
...
     public function getBasket()
     {
         return $this->basket;
     }
 
     /**
-     * Set basket.
-     *
-     * @param \Sonata\Component\Basket\BasketInterface $basket
+     * @param BasketInterface $basket
```
Comment (greg0ire) [inline src/BasketBundle/Entity/BaseBasketElement.php]: Can be removed

File: src/CustomerBundle/Entity/BaseAddress.php [hunk 3]
```diff
@@ -233,18 +232,14 @@ public function setFirstname($firstname): void
     }
 
     /**
-     * Get firstname.
-     *
-     * @return string
+     * {@inheritdoc}
```
Comment (greg0ire) [inline src/CustomerBundle/Entity/BaseAddress.php]: Can be removed

**Comment by ZRazor** [comment]
I close this one and will make a new one with docs fix only for 2.x and separate fixes for master.

**Pull request closed without merging**